frame/dvm/ethereum/src/lib.rs

```diff
@@ -309,6 +315,28 @@ impl<T: Trait> Module<T> {
 		frame_system::Module::<T>::deposit_log(digest.into());
 	}
 
+	/// Get the remaining balance for evm address
+	pub fn remaining_balance(account_id: &T::AccountId) -> T::Balance {
+		<RemainingBalance<T>>::get(account_id)
+	}
+
+	/// Inc remaining balance
+	pub fn inc_remain_balance(account_id: &T::AccountId, value: T::Balance) {
+		let remain_balance = Self::remaining_balance(account_id);
+		let updated_balance = remain_balance.saturating_add(value);
```

Comment:
There could be overflow too, should be gracely handled rather than ```saturating_add ```

Reply:
But these functions doesn't return a result which means it must be success.

Reply:
> There could be overflow too, should be gracely handled rather than `saturating_add `

Please check the latest commit again. What you said is absolutely right, when the sum of `remain_balance`  and  `param_value` over 10^9,  we should change the Currency balance value too.

An example:

balance A: 123_456_789_000_000_000_009
balance B: 100_000_000_000_999_999_999

then, A transfer to B 9 nano, the B balance result should be: `100_000_000_001_000_000_008`

It consists of two-part:
1. currency: 100_000_000_001
2. DVM remaining balance: 000_000_008

The overflow case in `inc(dec)_remain_balance` is handled in `mutate_account_basic`.